mcc/odin/odin/snapshot_list.go

```diff
@@ -41,6 +41,9 @@ func ListSnapshots(
 	err error,
 ) {
 	input := &rds.DescribeDBSnapshotsInput{}
+	if instanceName != "" {
+		input.SetDBInstanceIdentifier(instanceName)
+	}
```

Comment:
This is setting the instance name/id in the `DescribeDBSnapshotsInput`, so AWS RDS API will return only snapshots from this DBInstance. This is done only if the parameter is not empty.